src/cpu/testers/traffic_gen/gups_gen.cc

```diff
@@ -0,0 +1,261 @@
+#include "cpu/testers/traffic_gen/gups_gen.hh"
+
+#include <cstring>
+#include <string>
+
+#include "base/random.hh"
+#include "debug/GUPSGen.hh"
+
+#define POLY 0x0000000000000007UL
+#define PERIOD 1317624576693539401L
+
+
+GUPSGen::GUPSGen(const GUPSGenParams &params):
+    ClockedObject(params),
+    system(params.system),
+    requestorId(system->getRequestorId(this)),
+    nextSendEvent([this]{ sendNextBatch(); }, name()),
+    port(name() + ".port", this),
+    startAddr(params.start_addr),
+    memSize(params.mem_size),
+    blockSize(params.block_size)
+{}
+
+Port&
+GUPSGen::getPort(const std::string &if_name, PortID idx)
+{
+    if (if_name != "port"){
```

Comment:
style: Space after `)`

Please fix everywhere.

I've mentioned this before, but don't wait on style stuff to the end. Just do it right as you write the code.